### PR TITLE
[REVIEW] Add to_host specialization for strings in column test utilities

### PR DESCRIPTION
**Pull request by davidwendt**
This depends on #3278 
It creates a template specialization for `to_host` for strings columns.

### CODE REVIEW

File: cpp/tests/utilities/column_utilities.hpp [hunk 1]
```diff
@@ -18,37 +18,117 @@
...
+    CUDA_TRY(cudaMemcpy(host_bitmask.data(), c.null_mask(),
+                         mask_bytes, cudaMemcpyDeviceToHost));
+  }
+
+  for( size_type idx=0; idx < strings.size(); ++idx )
+  {
+      auto offset = h_offsets[idx];
+      auto length = h_offsets[idx+1] - offset;
+      auto mask_idx = idx / (sizeof(bitmask_type)*8);
+      if( !host_bitmask.empty() && (host_bitmask[mask_idx] & (1 << (idx % (sizeof(bitmask_type)*8))) > 0 ))
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/utilities/column_utilities.hpp]: So `bit_is_set` is a device function and this is CPU code on a host bitmask. Is there a host version of `bit_is_set`? Should we create one?
Also, probably not good practice to include a `.cuh` header in this `.hpp` file.

File: cpp/tests/utilities/column_utilities.hpp [hunk 2]
```diff
@@ -18,37 +18,117 @@
...
+    host_data.resize(col.size());
+    CUDA_TRY(cudaMemcpy(host_data.data(), col.view().head<T>(),
+                        col.size() * sizeof(T),
+                        cudaMemcpyDeviceToHost));
+  }
+
+  if (col.nullable()) {
+    size_t num_mask_elements =
+      std::ceil(static_cast<float>(mask_bytes) / sizeof(bitmask_type));
+    host_bitmask.resize(num_mask_elements);
```
Comment (jrhemstad) [inline cpp/tests/utilities/column_utilities.hpp]: https://github.com/rapidsai/cudf/pull/3282/files#diff-312a25454a338db96bb17c3e286afcd4R64
```suggestion
    host_bitmask.resize(num_bitmask_words(col.size()));
```

File: cpp/tests/utilities/column_utilities.hpp [hunk 3]
```diff
@@ -18,37 +18,117 @@
...
+/**
+ * @brief Copies the data and bitmask of a `column_view` of strings
+ * column to the host.
+ *
+ * @param c the `column_view` of strings to copy from
+ * @return std::pair first is `std::vector` of `std::string`
+ * and second is the column's bitmask.
+ */
+template <>
+inline std::pair<std::vector<std::string>, std::vector<bitmask_type>> to_host(column_view c) {
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/utilities/column_utilities.hpp]: This is a specialization of 
```
template <typename T>
std::pair<std::vector<T>, std::vector<bitmask_type>> to_host(column_view c) 
```
so we cannot change the parameter type.

File: cpp/tests/utilities/column_utilities.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrhemstad) [reply cpp/tests/utilities/column_utilities.hpp]: I'll be opening a PR shortly to address this.

File: cpp/tests/utilities/column_utilities.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrhemstad) [reply cpp/tests/utilities/column_utilities.hpp]: Ah yes, of course. I suppose just add a `CUDF_EXPECTS(c.type() == STRING)`

[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/utilities/column_utilities.hpp]: This line will make sure that it is a strings column
https://github.com/rapidsai/cudf/blob/abb5c6cae5108a33bc663d161e200a6381f52dec/cpp/tests/utilities/column_utilities.hpp#L105

File: cpp/tests/utilities/column_utilities.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrhemstad) [reply cpp/tests/utilities/column_utilities.hpp]: https://github.com/rapidsai/cudf/pull/3288

[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/utilities/column_utilities.hpp]: Removed this code logic since I don't think it is necessary for this function.

**Pull request merged**